docs/docs/examples.md

```diff
@@ -0,0 +1,317 @@
+# Signum Examples
+
+This page demonstrates how to accomplish common tasks using _Signum_.
+
+## Creating a Signed JSON Web Signature Object (`JwsSigned`)
+
+!!! info  inline end
+    This example requires the _Supreme_ KMP crypto provider and _Indispensable Josef_.
+
+In this example, we'll start with an ephemeral P-256 signer: 
+
+```kotlin
+val signer = Signer.Ephemeral {
+    ec { curve = ECCurve.SECP_256_R_1 }
+}.getOrThrow() //TODO handle error properly
+```
+
+Next up, we'll create a header and payload:
+
+```kotlin
+val header = JwsHeader(
+    algorithm = signer.signatureAlgorithm.toJwsAlgorithm().getOrThrow(),
+    jsonWebKey = signer.publicKey.toJsonWebKey()
+)
+val payload = byteArrayOf(1, 3, 3, 7)
+```
+
+Since both header and payload are fed into the signature, we need to prepare this signature input:
+
+```kotlin
+val plainSignatureInput = JwsSigned.prepareJwsSignatureInput(header, payload)
+```
+
+Now, everything is ready to be signed:
+
+```kotlin
+val signature = signer.sign(plainSignatureInput.encodeToByteArray()).signature //TODO: handle error
```

Comment:
should `prepareJwsSignatureInput` return `ByteArray`? I don't see a use case for it returning `String`

Reply:
Well, we're talking JSON here, so everything's a string. therefore, you need this string to construct the JwsSigned.

Reply:
yes, but a "signature input" is bytes; the fact that those bytes contain a json representation is an implementation detail; imo `prepare...SignatureInput` should return `ByteArray` instead of the caller having to explicitly do string -> bytes

Reply:
this is literally the string that is in the JWT. @nodh your take?